internal/app/machined/pkg/runtime/v1alpha1/bootloader/grub/grub.go

```diff
@@ -53,6 +54,9 @@ terminal_output console
 
 {{ range $label := .Labels -}}
 menuentry "{{ $label.Root }}" {
+{{ if .Version }}
```

Comment:
Needs to be conditional for now since it is possible that an entry doesn't have a version yet since all versions of Talos prior to the release that has this won't know to pass in the arg to specify which version it is.